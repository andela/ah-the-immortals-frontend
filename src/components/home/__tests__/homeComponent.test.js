import React from 'react';
import { Router } from 'react-router-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';
import Home from '../HomeComponent'; 


describe('Tests PasswordResetConfirmContainer', () => {
  const store = configureStore([thunk])({});
  const history = createBrowserHistory();
  const wrapper = mount(
    <Router history={history}>
      <Provider store={store}>
        <Home history={history} />
      </Provider>
    </Router>
  );
  it('Test mounting of component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
