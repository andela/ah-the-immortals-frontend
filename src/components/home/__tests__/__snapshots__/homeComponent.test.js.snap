// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests PasswordResetConfirmContainer Test mounting of component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Home
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div>
        <div
          className="background"
        >
          <Container
            className="homepage"
            fluid={false}
          >
            <div
              className="homepage container"
            >
              <HomePage>
                <div
                  className="columns"
                >
                  <Bootstrap(Row)>
                    <Row
                      bsPrefix="row"
                      noGutters={false}
                    >
                      <div
                        className="row"
                      >
                        <Col
                          className="col"
                        >
                          <div
                            className="col col"
                          >
                            <Container
                              fluid={false}
                            >
                              <div
                                className="container"
                              >
                                <br />
                                <br />
                                <h1
                                  className="homepage"
                                >
                                  Welcome to your articles Community
                                </h1>
                                <br />
                              </div>
                            </Container>
                            <Container
                              className="home"
                              fluid={false}
                            >
                              <div
                                className="home container"
                              >
                                <p>
                                  <b>
                                    Who is Authors Haven for?
                                  </b>
                                </p>
                                <p>
                                  Anyone looking to inspire and be inspired by ideas
                                </p>
                                <p>
                                  Join fellow authors and enthusiasts in Authors Haven
                                </p>
                              </div>
                            </Container>
                          </div>
                        </Col>
                      </div>
                    </Row>
                  </Bootstrap(Row)>
                </div>
              </HomePage>
            </div>
          </Container>
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-3 mt-4"
          >
            <TopRatedArticles>
              <div
                className="quotes position-fixed cardwidth cardmargins"
              >
                <div
                  className="box box1"
                >
                  <h5>
                    TOP RATED ARTICLES
                  </h5>
                  <hr />
                  <Link
                    to="#null"
                  >
                    <a
                      href="/#null"
                      onClick={[Function]}
                    >
                      Cras justo odio
                    </a>
                  </Link>
                  <hr />
                  <Link
                    to="#null"
                  >
                    <a
                      href="/#null"
                      onClick={[Function]}
                    >
                      Dapibus ac facilisis in
                    </a>
                  </Link>
                  <hr />
                  <Link
                    to="#null"
                  >
                    <a
                      href="/#null"
                      onClick={[Function]}
                    >
                      Vestibulum at eros
                    </a>
                  </Link>
                  <hr />
                  <Link
                    to="#null"
                  >
                    <a
                      href="/#null"
                      onClick={[Function]}
                    >
                      Vestibulum at eros
                    </a>
                  </Link>
                </div>
              </div>
            </TopRatedArticles>
          </div>
          <div
            className="col-md-6"
          >
            <Connect(Posts)>
              <Posts
                getPosts={[Function]}
                post={
                  Object {
                    "editpost": Array [],
                    "loading": false,
                    "post": Object {},
                    "posts": Array [],
                  }
                }
              >
                <div>
                  <div
                    className="post-feed"
                  >
                    <br />
                    <PostFeed
                      posts={Array []}
                    />
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="col"
                    />
                    <div
                      className="col"
                    >
                      <div>
                        <ul
                          className="pagination pagination-lg"
                        >
                          <li
                            className="page-item disabled"
                          >
                            <a
                              className="page-link"
                              href="#"
                            >
                              «
                            </a>
                          </li>
                          <li
                            className="page-item active"
                          >
                            <a
                              className="page-link"
                              href="#"
                            >
                              1
                            </a>
                          </li>
                          <li
                            className="page-item"
                          >
                            <a
                              className="page-link"
                              href="#"
                            >
                              2
                            </a>
                          </li>
                          <li
                            className="page-item"
                          >
                            <a
                              className="page-link"
                              href="#"
                            >
                              3
                            </a>
                          </li>
                          <li
                            className="page-item"
                          >
                            <a
                              className="page-link"
                              href="#"
                            >
                              4
                            </a>
                          </li>
                          <li
                            className="page-item"
                          >
                            <a
                              className="page-link"
                              href="#"
                            >
                              5
                            </a>
                          </li>
                          <li
                            className="page-item"
                          >
                            <a
                              className="page-link"
                              href="#"
                            >
                              »
                            </a>
                          </li>
                        </ul>
                      </div>
                    </div>
                    <div
                      className="col"
                    />
                  </div>
                  <br />
                  <br />
                  <br />
                  <br />
                </div>
              </Posts>
            </Connect(Posts)>
          </div>
          <div
            className="col-md-3 mt-4"
          >
            <PopularArticles>
              <div
                className="quotes position-fixed cardwidth"
              >
                <div
                  className="box box1"
                >
                  <h5>
                    POPULAR ARTICLES
                  </h5>
                  <hr />
                  <Link
                    to="#null"
                  >
                    <a
                      href="/#null"
                      onClick={[Function]}
                    >
                      Cras justo odio
                    </a>
                  </Link>
                  <hr />
                  <Link
                    to="#null"
                  >
                    <a
                      href="/#null"
                      onClick={[Function]}
                    >
                      Dapibus ac facilisis in
                    </a>
                  </Link>
                  <hr />
                  <Link
                    to="#null"
                  >
                    <a
                      href="/#null"
                      onClick={[Function]}
                    >
                      Vestibulum at eros
                    </a>
                  </Link>
                  <hr />
                  <Link
                    to="#null"
                  >
                    <a
                      href="/#null"
                      onClick={[Function]}
                    >
                      Vestibulum at eros
                    </a>
                  </Link>
                </div>
              </div>
            </PopularArticles>
          </div>
        </div>
      </div>
    </Home>
  </Provider>
</Router>
`;
